### ISSUE TITLE
Manually trigger periodic task using controller api.

### ISSUE THREAD

**Issue by amrishlal** [apache/incubator-pinot#7155]
I discussed this with @mcvsubbu a bit. Setup controller api to allow for manually triggering a periodic task. When the controller gets a request to manually execute a periodic task, it will send a broadcast helix message to all the controllers in the cluster for executing the periodic task against the tables. Upon getting the broadcast message, each controller will run the specified periodic task against the tables for which it is the lead controller. Lead controllers will then respond back to the originating controller with status information regarding periodic task execution which will be returned back to the user for information purposes. It will also be useful to allow for fetching status information of recent periodic tasks executions (without necessarily manually executing a periodic task), so that user can find out status of recent periodic task executions.

**Comment by Jackie-Jiang** [comment]
This is good design. So basically we want to guarantee that:
1. Only the leader controller can process the table
2. The API can be triggered on any controller and the request will be broadcasted to all the controllers

One challenge I can see here is that we need to gather the responses from the broadcasted helix message. We never use the custom Helix message this way, but the Helix messaging server does take an async callback stub, which we can probably leverage to achieve that.

**Comment by mcvsubbu** [comment]
Why gather responses? This is similar to our current segment reload message -- no response

**Comment by amrishlal** [comment]
> Helix messaging server does take an async callback stub, which we can probably leverage to achieve that.

async callback is what I was thinking about using.

> Why gather responses? This is similar to our current segment reload message -- no response

if one of the controllers fails to complete manually triggered periodic task successfully, then the user may want to know about that and hence the need for gathering responses. Also, user may want status information on when the periodictask was run last and what was the result of that run before deciding to trigger the periodictask manually again.

Also it is interesting to note that PinotTaskManager (which is derived from ControllerPeriodTask) already seems to have an extensive set of controller apis (REST path "/tasks" through PinotTaskRestletResource) for adding/removing/deleting tasks and getting their status information, but "PeriodicTask" is a higher level construct, so I was thinking about adding another rest path "/periodictask" for triggering and getting status of PeriodicTask derived classes (excluding PinotTaskManager).

Another option may be to deprecate current "/tasks" REST path and create "/periodictask" and "/periodictask/taskmanager" REST paths where the REST path "/periodictask/taskmanager" would be same as current "/tasks" REST path?

**Comment by mcvsubbu** [comment]
No, let us not complicate this any more than we have to. Unless helix provides a response option (I think it does not). The user can trigger the API and scan the logs for further info. In any case, the logs will have more detailed information on what went wrong.
